### PR TITLE
feat: Setup workspace workflow file.

### PR DESCRIPTION
**Pull request by yubonluo**
### Description

We are going to onboard a new dashboards plugin dashboards-workspace to FTRepo, this is the initial PR to setup basic workflow file. Test cases will be included in following PRs.

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/workspace-release-e2e-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      test-name: dashboards workspace
+      test-command: env CYPRESS_DASHBOARDS_WORKSPACE_ENABLED=true yarn cypress:run-with-security --browser chromium --spec 'cypress/integration/plugins/dashboards-workspace/*'
+      osd-serve-args: --workspace.enabled=true --savedObjects.permission.enabled=true
+  tests-without-security:
+    needs: changes
+    if: ${{ needs.changes.outputs.tests == 'true' }}
+    uses: ./.github/workflows/release-e2e-workflow-template.yml
+    with:
+      test-name: dashboards workspace
+      test-command: env CYPRESS_DASHBOARDS_WORKSPACE_ENABLED=true yarn cypress:run-with-security --browser chromium --spec 'cypress/integration/plugins/dashboards-workspace/*'
```
Comment (Hailong-am) [inline .github/workflows/workspace-release-e2e-workflow.yml:32]: should be `cypress:run-without-security`?

File: .github/workflows/workspace-release-e2e-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      osd-serve-args: --workspace.enabled=true --savedObjects.permission.enabled=true
```
Comment (Hailong-am) [inline .github/workflows/workspace-release-e2e-workflow.yml:33]: `--savedObjects.permission.enabled=false` for without security case

File: cypress/integration/plugins/dashboards-workspace/workspace_create.spec.js
Comment (ruanyl) [inline cypress/integration/plugins/dashboards-workspace/workspace_create.spec.js:1]: Should we move this file to `cypress/integration/core-opensearch-dashboards/opensearch-dashboards` as workspace is a plugin in core repo.

File: cypress.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
     "DATASOURCE_MANAGEMENT_ENABLED": false,
     "ML_COMMONS_DASHBOARDS_ENABLED": true,
     "WAIT_FOR_LOADER_BUFFER_MS": 0,
-    "DASHBOARDS_ASSISTANT_ENABLED": false
+    "DASHBOARDS_ASSISTANT_ENABLED": false,
+    "DASHBOARDS_WORKSPACE_ENABLED": false
```
Comment (wanglam) [inline cypress.json:25]: Shall we need start with `DASHBOARDS`? I think workspace is not direct related dashboard.

Comment (yubonluo) [reply cypress.json:25]: Sure, I will delete the prefix `DASHBOARDS `.

File: cypress/integration/plugins/dashboards-workspace/workspace_create.spec.js
Comment (yubonluo) [reply cypress/integration/plugins/dashboards-workspace/workspace_create.spec.js:1]: Yes, I will move the integration test file to `cypress/integration/core-opensearch-dashboards/opensearch-dashboards`.

File: .github/workflows/workspace-release-e2e-workflow.yml (hunk 2 above)
Comment (yubonluo) [reply .github/workflows/workspace-release-e2e-workflow.yml:33]: I will correct it.

File: .github/workflows/workspace-release-e2e-workflow.yml (hunk 1 above)
Comment (yubonluo) [reply .github/workflows/workspace-release-e2e-workflow.yml:32]: Yes, I will correct it.

File: .github/workflows/workspace-release-e2e-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      tests: ${{ steps.filter.outputs.tests }}
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            tests:
+              - 'cypress/**/dashboards-workspace/**'
```
Comment (SuZhou-Joe) [inline .github/workflows/workspace-release-e2e-workflow.yml:17]: The test pattern should match `cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin` I think?

**Review by SuZhou-Joe — COMMENTED** [review]
Nice setup for workspace plugin! Left a comment

**Review by wanglam — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**